go/registry/api/api.go

```diff
@@ -888,9 +888,8 @@ func VerifyRegisterRuntimeArgs(logger *logging.Logger, sigRt *SignedRuntime, isG
 		return nil, ErrInvalidArgument
 	}
 
-	if !isGenesis && !rt.Genesis.StateRoot.IsEmpty() {
```

Comment:
Are you sure this same check is covered in `SanityCheck`? So if in non-genesis, registrations which use a non-empty `StateRoot` must be rejected until we have a resolution to #1686.

Reply:
Yeah, `SanityCheck()` only covers the cases specified in `RuntimeGenesis` struct. I put the `!isGenesis && !rt.Genesis.StateRoot.IsEmpty()` back in now and a reference to #1686 .